File: clients/python/examples/collections-find-many-sort-vector.py
from astrapy import DataAPIClient

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Find documents
cursor = collection.find(
    {}, sort={"$vectorize": "Text to vectorize"}, include_sort_vector=True
)

# Get the sort vector from the result
vector = cursor.get_sort_vector()

print(vector)

File: clients/csharp/examples/collections-find-many-sort-vector.cs
using System.Text.Json;
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Collections;
using DataStax.AstraDB.DataApi.Core;
using DataStax.AstraDB.DataApi.Core.Query;

namespace Examples;

public class Program
{
  static void Main()
  {
    // Get an existing collection
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );
    var collection = database.GetCollection("**COLLECTION_NAME**");

    // Find documents
    var result = collection.Find(
      new CollectionFindOptions<Document>()
      {
        Sort = Builders<Document>.CollectionSort.Vectorize(
          "Text to vectorize"
        ),
        IncludeSortVector = true,
      }
    );

    // Inspect the sort vector
    Console.WriteLine(JsonSerializer.Serialize(result.GetSortVector()));
  }
}
